### PR TITLE
Updated/added keychain functions for macOS Sierra

### PR DESCRIPTION
**Pull request by harrywynn**
macOS Sierra requires an additional keychain command of set-key-partition-list after all certificates are imported.  This update will automatically detect if the node is running 10.12 or later and perform this command.

### CODE REVIEW

File: src/main/java/au/com/rayh/DeveloperProfileLoader.java [hunk 1]
```diff
@@ -79,12 +107,31 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
             args = new ArgumentListBuilder("security","import");
             args.add(id).add("-k",keyChain);
             args.add("-P").addMasked(dp.getPassword().getPlainText());
-            args.add("-T","/usr/bin/codesign");
-            args.add("-T","/usr/bin/productsign");
+            
+            // flag is different depending on OS
+            if (setKeyPartitionList) {
+	            args.add("-A","/usr/bin/codesign");
```
Comment (Brantone) [inline src/main/java/au/com/rayh/DeveloperProfileLoader.java]: Doesn't `-A` mean "allow any", thus it doesn't require a value.

File: src/main/java/au/com/rayh/DeveloperProfileLoader.java [hunk 2]
```diff
@@ -79,12 +107,31 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
...
+	            args.add("-T","/usr/bin/productsign");
+	        }
+	        
             args.add(keyChain);
             invoke(launcher, listener, args, "Failed to import identity "+id);
         }
 
+
+		// necessary for functional code signing on macOS Sierra and up
+        if (setKeyPartitionList) {
```
Comment (Brantone) [inline src/main/java/au/com/rayh/DeveloperProfileLoader.java]: Something is off with the spacing here.

**Comment by Brantone** [comment]
This implementation seems like bit of a hidden feature, dependent on an arbitrary environment variable. Surely there has to be a way to auto-detect the platform or even implement as a check-box instead?

**Comment by harrywynn** [comment]
@Brantone agreed, the environment variable was a hacky way to get it working.  I've updated it to automatically detect the OS version and do the proper thing.

File: src/main/java/au/com/rayh/DeveloperProfileLoader.java [hunk 3]
```diff
@@ -5,9 +5,11 @@
 import hudson.Extension;
 import hudson.FilePath;
 import hudson.Launcher;
+import hudson.EnvVars;
```
Comment (aheritier) [inline src/main/java/au/com/rayh/DeveloperProfileLoader.java]: useless import @harrywynn now

Comment (harrywynn) [inline src/main/java/au/com/rayh/DeveloperProfileLoader.java]: @aheritier unnecessary import removed

**Comment by aheritier** [comment]
LGTM but may requires more tests

**Comment by BennyHarv3** [comment]
I don't know what the status of this pull request is anymore, but playing around with the Mac keychain for Jenkins, I noticed that you don't need the -A part of the security call that you have [around line 108](https://github.com/jenkinsci/xcode-plugin/pull/77/files?diff=unified#diff-45ab7459f7c7a1253e287f7a24a09d4eR108) of the new file. Using the -T options works in Sierra for me. I do still need the set-key-partition-list, though.

**Pull request closed without merging**